libs/dao/src/main/java/com/akto/dao/testing/DeleteRunsInfoDao.java

```diff
@@ -0,0 +1,88 @@
+package com.akto.dao.testing;
+
+import java.util.List;
+import java.util.Map;
+
+import org.bson.conversions.Bson;
+import org.bson.types.ObjectId;
+
+import com.akto.dao.AccountsContextDao;
+import com.akto.dao.testing_run_findings.TestingRunIssuesDao;
+import com.akto.dto.test_run_findings.TestingRunIssues;
+import com.akto.dto.testing.DeleteRunsInfo;
+import com.akto.dto.testing.TestingRunResult;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+
+public class DeleteRunsInfoDao extends AccountsContextDao<DeleteRunsInfo> {
+
+    public static final DeleteRunsInfoDao instance = new DeleteRunsInfoDao();
+
+    private static Map<String, Boolean> deleteTestRunsCollections(List<ObjectId> testRunIds,
+            List<ObjectId> latestSummaryIds, List<Integer> testConfigIds,
+            Map<String, Boolean> testingCollectionsDeleted) {
+                
+        if (!testingCollectionsDeleted.getOrDefault("testing_run_result", false)) {
+            TestingRunResultDao.instance.deleteAll(Filters.in(TestingRunResult.TEST_RUN_RESULT_SUMMARY_ID, latestSummaryIds));
+            testingCollectionsDeleted.put("testing_run_result", true);
+        }
+
+        if (!testingCollectionsDeleted.getOrDefault("testing_run_issues", false)) {
+            TestingRunIssuesDao.instance
+                    .deleteAll(Filters.in(TestingRunIssues.LATEST_TESTING_RUN_SUMMARY_ID, latestSummaryIds));
+            testingCollectionsDeleted.put("testing_run_issues", true);
+        }
+
+        if (!testingCollectionsDeleted.getOrDefault("testing_run_result_summaries", false)) {
+            TestingRunResultSummariesDao.instance
+                    .deleteAll(Filters.in("_id", latestSummaryIds));
+            testingCollectionsDeleted.put("testing_run_result_summaries", true);
+        }
+
+        if (!testingCollectionsDeleted.getOrDefault("testing_run_config", false)) {
+            TestingRunConfigDao.instance.deleteAll(Filters.in("_id", testConfigIds));
+            testingCollectionsDeleted.put("testing_run_config", true);
+        }
+
+        if (!testingCollectionsDeleted.getOrDefault("testing_run", false)) {
```

Comment:
testRunIds is being passed as [] if summary is deleted. Can we avoid executing this query in that case?

Reply:
For the cases of Ci/Cd and recurring, we dont want to delete test runs and its config. So in that case we use summaryIds to delete. But to use the same function, i passed testrunids and configids as empty. so there'll be no document found in mongo